### ISSUE TITLE
[Bug]: snaps Installation: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'xxx' for key 'oc_polls_options.UNIQ_options'"

### ISSUE THREAD

**Issue by kratzerph** [nextcloud/polls#2853]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Hi,

hoping this is the right place to address my problem. 
We are using snaps with automatic upgrades enabled. Last night snaps updated our nextcloud from Version 25.0.4 to 25.0.5. Looks like that nextcloud is now hanging between maintenance mode and normal mode.  Currently nextcloud has high cpu load:

see Nextcloud logs

### Steps to reproduce

Can't reproduce

### Expected behavior

Upgrade finishes without problems....

### Installation method

Community SNAP package

### Nextcloud Server version

25

### Operating system

Debian/Ubuntu

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 22.2.3 to 22.2.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [X] SSO - SAML
- [ ] Other

### Configuration report

```shell
extcloud.occ config:list system
Nextcloud is in maintenance mode, hence the database isn't accessible.
Cannot perform any command except 'maintenance:mode --off'

{
    "system": {
        "apps_paths": [
            {
                "path": "\/snap\/nextcloud\/current\/htdocs\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/snap\/nextcloud\/current\/nextcloud\/extra-apps",
                "url": "\/extra-apps",
                "writable": true
            }
        ],
        "supportedDatabases": [
            "mysql"
        ],
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - admin_audit: 1.15.0
  - appointments: 1.14.14
  - bruteforcesettings: 2.5.0
  - calendar: 4.3.2
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - collectives: 2.4.0
  - comments: 1.15.0
  - contacts: 5.2.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - deck: 1.8.3
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - forms: 3.2.0
  - integration_github: 1.0.15
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - mail: 2.2.5
  - metadata: 0.18.0
  - nextcloud_announcements: 1.14.0
  - notes: 4.7.2
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - onlyoffice: 7.8.0
  - password_policy: 1.15.0
  - photos: 2.0.1
  - polls: 5.0.0
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - tasks: 0.14.5
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - user_saml: 5.1.2
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
```

### Nextcloud Logs

```shell
`\UniqueConstraintViolationException","Message":"An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '36-6095e3973c77d9e5e5774c733391f68d-1665439200' for key 'oc_polls_options.UNIQ_options'","Code":1062,"Trace":[{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1780,"function":"convert","class":"Doctrine\\DBAL\\Driver\\API\\MySQL\\ExceptionConverter","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1719,"function":"handleDriverException","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1067,"function":"convertExceptionDuringQuery","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Connection.php","line":261,"function":"executeQuery","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1809,"function":"executeQuery","class":"OC\\DB\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Migrator.php","line":162,"function":"query","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Migrator.php","line":69,"function":"applySchema","class":"OC\\DB\\Migrator","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Connection.php","line":582,"function":"migrate","class":"OC\\DB\\Migrator","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/ConnectionAdapter.php","line":237,"function":"migrateToSchema","class":"OC\\DB\\Connection","type":"->"},{"file":"/var/snap/nextcloud/34073/nextcloud/extra-apps/polls/lib/Db/IndexManager.php","line":48,"function":"migrateToSchema","class":"OC\\DB\\ConnectionAdapter","type":"->"},{"file":"/var/snap/nextcloud/34073/nextcloud/extra-apps/polls/lib/Migration/RepairSteps/CreateIndices.php","line":48,"function":"migrate","class":"OCA\\Polls\\Db\\IndexManager","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/Repair.php","line":127,"function":"run","class":"OCA\\Polls\\Migration\\RepairSteps\\CreateIndices","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/core/Command/Maintenance/Repair.php","line":123,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Command/Command.php","line":255,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":1009,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":273,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":149,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/Console/Application.php","line":214,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/occ","line":11,"args":["/snap/nextcloud/34073/htdocs/console.php"],"function":"require_once"}],"File":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php","Line":60,"Previous":{"Exception":"Doctrine\\DBAL\\Driver\\PDO\\Exception","Message":"SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '36-6095e3973c77d9e5e5774c733391f68d-1665439200' for key 'oc_polls_options.UNIQ_options'","Code":1062,"Trace":[{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Driver/PDO/Connection.php","line":87,"function":"new","class":"Doctrine\\DBAL\\Driver\\PDO\\Exception","type":"::"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1062,"function":"query","class":"Doctrine\\DBAL\\Driver\\PDO\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Connection.php","line":261,"function":"executeQuery","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1809,"function":"executeQuery","class":"OC\\DB\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Migrator.php","line":162,"function":"query","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Migrator.php","line":69,"function":"applySchema","class":"OC\\DB\\Migrator","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Connection.php","line":582,"function":"migrate","class":"OC\\DB\\Migrator","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/ConnectionAdapter.php","line":237,"function":"migrateToSchema","class":"OC\\DB\\Connection","type":"->"},{"file":"/var/snap/nextcloud/34073/nextcloud/extra-apps/polls/lib/Db/IndexManager.php","line":48,"function":"migrateToSchema","class":"OC\\DB\\ConnectionAdapter","type":"->"},{"file":"/var/snap/nextcloud/34073/nextcloud/extra-apps/polls/lib/Migration/RepairSteps/CreateIndices.php","line":48,"function":"migrate","class":"OCA\\Polls\\Db\\IndexManager","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/Repair.php","line":127,"function":"run","class":"OCA\\Polls\\Migration\\RepairSteps\\CreateIndices","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/core/Command/Maintenance/Repair.php","line":123,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Command/Command.php","line":255,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":1009,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":273,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":149,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/Console/Application.php","line":214,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/occ","line":11,"args":["/snap/nextcloud/34073/htdocs/console.php"],"function":"require_once"}],"File":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Driver/PDO/Exception.php","Line":26,"Previous":{"Exception":"PDOException","Message":"SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '36-6095e3973c77d9e5e5774c733391f68d-1665439200' for key 'oc_polls_options.UNIQ_options'","Code":"23000","Trace":[{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Driver/PDO/Connection.php","line":82,"function":"query","class":"PDO","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1062,"function":"query","class":"Doctrine\\DBAL\\Driver\\PDO\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Connection.php","line":261,"function":"executeQuery","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Connection.php","line":1809,"function":"executeQuery","class":"OC\\DB\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Migrator.php","line":162,"function":"query","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Migrator.php","line":69,"function":"applySchema","class":"OC\\DB\\Migrator","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/Connection.php","line":582,"function":"migrate","class":"OC\\DB\\Migrator","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/DB/ConnectionAdapter.php","line":237,"function":"migrateToSchema","class":"OC\\DB\\Connection","type":"->"},{"file":"/var/snap/nextcloud/34073/nextcloud/extra-apps/polls/lib/Db/IndexManager.php","line":48,"function":"migrateToSchema","class":"OC\\DB\\ConnectionAdapter","type":"->"},{"file":"/var/snap/nextcloud/34073/nextcloud/extra-apps/polls/lib/Migration/RepairSteps/CreateIndices.php","line":48,"function":"migrate","class":"OCA\\Polls\\Db\\IndexManager","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/Repair.php","line":127,"function":"run","class":"OCA\\Polls\\Migration\\RepairSteps\\CreateIndices","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/core/Command/Maintenance/Repair.php","line":123,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Command/Command.php","line":255,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":1009,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":273,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/3rdparty/symfony/console/Application.php","line":149,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/lib/private/Console/Application.php","line":214,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/snap/nextcloud/34073/htdocs/occ","line":11,"args":["/snap/nextcloud/34073/htdocs/console.php"],"function":"require_once"}],"File":"/snap/nextcloud/34073/htdocs/3rdparty/doctrine/dbal/src/Driver/PDO/Connection.php","Line":82}}},"message":"Exception while executing repair step Polls - Create indices and foreign key constraints","exception":{},"CustomMessage":"Exception while executing repair step Polls - Create indices and foreign key constraints"}}`
```

### Additional info

root        1050  0.1  0.2 1318588 42380 ?       Ssl  Mär29  37:48 /usr/lib/snapd/snapd
root       31631  0.0  0.0   4644  1784 ?        Ss   03:42   0:00 /bin/sh /snap/nextcloud/34073/bin/start-php-fpm
root       31633  0.0  0.0   4644  1700 ?        Ss   03:42   0:00 /bin/sh /snap/nextcloud/34073/bin/start-redis-server
root       31635  0.0  0.0   4644  1748 ?        Ss   03:42   0:05 /bin/sh -e /snap/nextcloud/34073/bin/start_mysql
root       31637  0.0  0.0   4644  1728 ?        Ss   03:42   0:00 /bin/sh /snap/nextcloud/34073/bin/renew-certs
root       31639  0.0  0.0   4644   864 ?        Ss   03:42   0:00 /bin/sh /snap/nextcloud/34073/bin/delay-on-failure mdns-publisher nextcloud
root       31641  0.0  0.0   4644  1680 ?        Ss   03:42   0:06 /bin/sh /snap/nextcloud/34073/bin/nextcloud-cron
root       31645  0.0  0.0   4644  1676 ?        Ss   03:42   0:00 /bin/sh /snap/nextcloud/34073/bin/run-httpd -k start -DFOREGROUND
root       32084  0.0  0.0   4644  1736 ?        S    03:42   0:00 /bin/sh /snap/nextcloud/34073/bin/mysqld_safe --defaults-file=/snap/nextcloud/34073/my.cnf --datadir=/var/snap/nextcloud/34073/mysql --pid-file=/tmp/pids/mysql.pid --lc-messages-dir=/snap/nextcloud/34073/share --socket=/tmp/sockets/mysql.sock
root       32321 99.5  2.3 2691712 381888 ?      Sl   03:42 472:02 /snap/nextcloud/34073/bin/mysqld --defaults-file=/snap/nextcloud/34073/my.cnf --basedir=/snap/nextcloud/34073 --datadir=/var/snap/nextcloud/34073/mysql --plugin-dir=/snap/nextcloud/34073/lib/plugin --lc-messages-dir=/snap/nextcloud/34073/share --log-error=../logs/mysql_errors.log --pid-file=/tmp/pids/mysql.pid --socket=/tmp/sockets/mysql.sock
root       32657  0.0  0.2 342220 34048 ?        Ss   03:42   0:02 php-fpm: master process (/snap/nextcloud/34073/config/php/php-fpm.conf)
root       37639  0.0  0.0   4644  1748 ?        S    03:43   0:00 /bin/sh /snap/nextcloud/34073/bin/httpd-wrapper -k start -DFOREGROUND
root       37696  0.0  0.0  43884  9332 ?        S    03:43   0:01 httpd -d /snap/nextcloud/34073 -DEnableHTTPS -DEnableHSTS -k start -DFOREGROUND
root       37705  0.0  0.0 1982860 15508 ?       Sl   03:43   0:17 httpd -d /snap/nextcloud/34073 -DEnableHTTPS -DEnableHSTS -k start -DFOREGROUND
root       37706  0.0  0.0 1983152 15916 ?       Sl   03:43   0:24 httpd -d /snap/nextcloud/34073 -DEnableHTTPS -DEnableHSTS -k start -DFOREGROUND
root       37707  0.0  0.0 1983444 16204 ?       Sl   03:43   0:20 httpd -d /snap/nextcloud/34073 -DEnableHTTPS -DEnableHSTS -k start -DFOREGROUND
root       40553  0.1  0.0 1983232 15776 ?       Sl   03:44   0:28 httpd -d /snap/nextcloud/34073 -DEnableHTTPS -DEnableHSTS -k start -DFOREGROUND
root     1746503  0.1  0.0 1981328 13232 ?       Sl   10:42   0:04 httpd -d /snap/nextcloud/34073 -DEnableHTTPS -DEnableHSTS -k start -DFOREGROUND
root     1978133  0.2  0.0   4644  1708 ?        Ss   11:36   0:00 /bin/sh -e /snap/nextcloud/34073/bin/nextcloud-fixer
root     1978322  0.0  0.0   4536   820 ?        S    11:36   0:00 run-parts -v --exit-on-error --regex .*\.sh$ /snap/nextcloud/34073/fixes/existing-install
root     1978925  0.0  0.0   4644   856 ?        S    11:36   0:00 /bin/sh -e /snap/nextcloud/34073/fixes/existing-install/5_regenerate-assets.sh
root     1978926  0.0  0.0   4644  1804 ?        S    11:36   0:00 /bin/sh -e /snap/nextcloud/34073/bin/occ -n maintenance:repair
root     1978966  0.0  0.0   4644  1744 ?        S    11:36   0:00 /bin/sh /snap/nextcloud/34073/bin/run-php /snap/nextcloud/34073/htdocs/occ -n maintenance:repair
root     1978989 23.8  0.4 250688 74296 ?        S    11:36   0:01 php -c /snap/nextcloud/34073/config/php /snap/nextcloud/34073/htdocs/occ -n maintenance:repair

**Comment by dartcafe** [comment]
@kratzerph There are duplicate entries in the options database, which should not be because duplicates get removed in a pre migration repair step. 

Are you able to send me a db dump? 
Relevant are the columns `poll_id`, `poll_option_text`, `timestamp`, `poll_option_hash`.

**Comment by kratzerph** [comment]
hi dartcafe,

thanks for your reply. right now the problem seemed to get fixed(maybe just temporary till next update) by disabling the poll app, waiting for a complete run of the upgrade scripts, and then reenable poll app...

if you tell me(or just give me a link) how to dump / look in the snaps mysql tables, i can give you needed information....

greets

**Comment by dartcafe** [comment]
Just login to your mySQL console and export the contents of the `oc_polls_options` table or use any MySQL frontend.

**Comment by dartcafe** [comment]
Was it a fisrt time install of polls?

**Comment by kratzerph** [comment]
@dartcafe : i never used polls, but i guess that some other users on our instance use polls, see timestamps of rows. last days a new snap of nextcloud was released, 25.0.5snap2. This update resulted again in an update loop. My collegue disabled polls app and now it's fine again....maybe till next update :-)

**Comment by dartcafe** [comment]
If all works as it should, duplicates get removed from the databases where necessary, in a pre-migration step before any migration or other db activities. If nextcloud thinks it is a first time install, the pre-migration steps get skipped. Possibly disabling and enabling Polls did the correct job. 

An update of Polls will land to the app store soon. You may come back here if this occurs again or close the issue, if not.

I will check the migrations anyways.
